run-tests/src/utils.js

```diff
@@ -162,21 +165,75 @@ class TestRunner {
               clearInterval(poller);
               TestRunner._parseApiResult(content);
               this.build_status = content.status;
-              resolve();
+              resolve(content);
             }
           }
         });
       }, WATCH_INTERVAL);
     });
   }
 
+  static async _uploadResults(content) {
+    const promises = [];
+    core.info(`Uploading test report to artifacts for build id: ${content.id}`);
+    const { devices, id: buildId, framework } = content;
+    const rootDir = './reports';
+    const username = process.env[ENV_VARS.BROWSERSTACK_USERNAME].replace("-GitHubAction", "");
+    const accesskey = process.env[ENV_VARS.BROWSERSTACK_ACCESS_KEY];
+    for (const device of devices) {
+      const { sessions } = device;
+      for (const session of sessions) {
+        const { id } = session;
+        const options = {
+          url: `https://${username}:${accesskey}@${URLS.BASE_URL}/${URLS.WATCH_FRAMEWORKS[framework]}/${buildId}/sessions/${id}/${URLS.REPORT[framework]}`,
+        };
+        /* eslint-disable no-eval */
+        promises.push(new Promise((resolve, reject) => {
+          request.get(options, (error, response) => {
+            if (error) {
+              reject(error);
+            }
+            if (response.statusCode !== 200) {
+              reject(response.body);
+            }
+            resolve(response.body);
+          });
+        }).then(async (report) => {
+          if (!fs.existsSync(rootDir)) {
+            fs.mkdirSync(rootDir);
+          }
+          if (framework === FRAMEWORKS.espresso) {
```

Comment:
Lets add to readme that we only support junit xml for espresso and resultBundle for xcuitest

Reply:
Also check we if support Junit XML report with cucumber

Reply:
https://www.browserstack.com/docs/app-automate/espresso/run-cucumber-tests

Reply:
Added support for cucumber by checking is cucumberOptions.plugins is present in the input_capabilities